src/parsers/manifest/dash/parse_representations.ts

```diff
@@ -29,8 +29,49 @@ import {
   IRepresentationIntermediateRepresentation,
 } from "./node_parsers/Representation";
 import { IParsedSegmentTemplate } from "./node_parsers/SegmentTemplate";
+import { IScheme } from "./node_parsers/utils";
 import parseRepresentationIndex from "./parse_representation_index";
 
+/**
+ * Combine unique inband event streams from representation and
+ * adaptation data.
+ * @param {Object} representation
+ * @param {Object} adaptation
+ * @returns {undefined | Array.<Object>}
+ */
+function getAcceptedInbandEventStreams(
+  representation: IRepresentationIntermediateRepresentation,
+  adaptation: IAdaptationSetIntermediateRepresentation
+): IScheme[] | undefined {
+  const newSchemeId = [];
+  if (representation.children.signaledInbandEventSchemeIds !== undefined) {
+    newSchemeId.push(...representation.children.signaledInbandEventSchemeIds);
+  }
+  if (adaptation.children.signaledInbandEventSchemeIds !== undefined) {
+    if (newSchemeId.length === 0) {
+      newSchemeId.push(...adaptation.children.signaledInbandEventSchemeIds);
+    } else {
+      const len = adaptation.children.signaledInbandEventSchemeIds.length;
+      for (let i = 0; i < len; i++) {
+        const signaledInbandEventSchemeId =
+          adaptation.children.signaledInbandEventSchemeIds[i];
+        const isNonDuplicatedInbandEventStream =
+          !newSchemeId.some(({ schemeIdUri, value }) => {
```

Comment:
Aren't you checking between event streams coming from an Adaptation here?
Why not, but you don't do it if there's no event stream in a Representation (`if (newSchemeId.length === 0) {`) so it doesn't make much sense.

Reply:
You're right, I check duplicates here only for adaptations inbandeventstream. I intended to just not duplicate when it exists in rep inbandeventstream. I have to change that